destination/franz.go

```diff
@@ -61,41 +89,91 @@ func NewFranzProducer(ctx context.Context, cfg Config) (*FranzProducer, error) {
 	return &FranzProducer{
 		client:     cl,
 		keyEncoder: keyEncoder,
+		getTopic:   topicFn,
 	}, nil
 }
 
 func (p *FranzProducer) Produce(ctx context.Context, records []sdk.Record) (int, error) {
-	var (
-		wg      sync.WaitGroup
-		results = make(kgo.ProduceResults, 0, len(records))
-		promise = func(r *kgo.Record, err error) {
-			results = append(results, kgo.ProduceResult{Record: r, Err: err})
-			wg.Done()
+	if len(records) == 1 {
+		// Fast path for a single record.
+		rec, err := p.prepareRecord(records[0])
+		if err != nil {
+			return 0, fmt.Errorf("failed to prepare record: %w", err)
 		}
+		_, err = p.client.ProduceSync(ctx, rec).First()
```

Comment:
what's the advantage of doing `ProduceSync` vs `Produce` when there are more than one single record? I'd like to understand what's the benefit of having two different paths for 1 record vs more than 1.

Reply:
It doesn't do unnecessary allocations and spawn additional goroutines. I didn't benchmark it, but I'd expect it to be faster.

The thing is that by default the connector will be writing records in a batch of 1, so I'm trying to make the default experience as good as possible.

Reply:
@lovromazgon got it. That makes sense!